x-pack/plugins/enterprise_search/server/routes/app_search/credentials.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { IRouteDependencies } from '../../plugin';
+import { createEnterpriseSearchRequestHandler } from '../../lib/enterprise_search_request_handler';
+
+interface ICredentialsResponse {
+  results: object[];
+  meta?: {
+    page?: {
+      total_results: number;
+    };
+  };
+}
+
+export function registerCredentialsRoutes({ router, config, log }: IRouteDependencies) {
+  router.get(
+    {
+      path: '/api/app_search/credentials',
+      validate: {
+        query: schema.object({
+          'page[current]': schema.number(),
```

Comment:
I wonder if we should standardize this. EnginesOverview uses `pageIndex` for example, and I'll admit it is a little odd in this specific scenario to use only `page.current` when we don't have a `page.anythingElse`